cpp/src/arrow/compute/row/grouper.cc

```diff
@@ -448,17 +431,6 @@ struct AnyKeysSegmenter : public BaseRowSegmenter {
   group_id_t save_group_id_;
 };
 
-Status CheckAndCapLengthForConsume(int64_t batch_length, int64_t& consume_offset,
```

Comment:
Moved down into the next anonymous namespace enclosing `Grouper` because this function is for `Grouper` to use.